pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go

```diff
@@ -67,12 +129,14 @@ func (ac *OSSAccessControlService) GetUserPermissions(ctx context.Context, user
 	builtinRoles := ac.GetUserBuiltInRoles(user)
 	permissions := make([]*accesscontrol.Permission, 0)
 	for _, builtin := range builtinRoles {
-		if roleNames, ok := accesscontrol.FixedRoleGrants[builtin]; ok {
+		if values, ok := accesscontrol.FixedRoleGrants.Load(builtin); ok {
+			roleNames := values.([]string)
 			for _, name := range roleNames {
-				r, exists := accesscontrol.FixedRoles[name]
+				value, exists := accesscontrol.FixedRoles.Load(name)
 				if !exists {
 					continue
 				}
+				r := value.(accesscontrol.RoleDTO)
```

Comment:
We use `sync.Map` in this case to have safe roles storage. But it forces us to use reflections and loose some type safety. Is it better to build struct containing map and use locking primitive for safe access? In this case we have better types. [`sunc.Map` docs](https://golang.org/pkg/sync/#Map) clearly says that it's for some rare cases and I don't think we need that potential performance boost it can provide in exchange of of type safety.

Reply:
Hey @alexanderzobnin, excellent feedback and really like that you took time in explaining :+1: 
I will make said changes ! in other words, use Map + Lock